lib/rtp_h264/depayloader.ex

```diff
@@ -0,0 +1,118 @@
+defmodule Membrane.Element.RTP.H264.Depayloader do
+  @moduledoc """
+  Depayloads H264 RTP payloads into H264 NAL Units.
+  """
+  use Membrane.Element.Base.Filter
+  use Bunch
+  use Membrane.Log
+
+  alias Membrane.Buffer
+  alias Membrane.Caps.{RTP, Video.H264}
+  alias Membrane.Event.Discontinuity
+  alias Membrane.Element.RTP.H264.{FU, NALHeader, StapA}
+  alias NALHeader.PayloadTypeDecoder
+
+  @frame_prefix <<1::32>>
+  @type sequence_number :: 0..65_535
+
+  def_output_pads output: [
+                    caps: {H264, stream_format: :byte_stream}
+                  ]
+
+  def_input_pads input: [
+                   caps: {RTP, payload_type: :dynamic},
+                   demand_unit: :buffers
+                 ]
+
+  defmodule State do
+    @moduledoc false
+    # pp_acc, stands for packet parser accumulator
+    defstruct pp_acc: nil
+  end
+
+  def handle_init(_) do
+    {:ok, %State{}}
+  end
+
+  @impl true
+  def handle_caps(:input, _caps, _context, state) do
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_process(_pad, %Buffer{payload: payload} = buffer, _ctx, state) do
+    NALHeader.parse_unit_header(payload)
+    ~>> ({:ok, {header, rest}} ->
+           case PayloadTypeDecoder.decode_type(header.type) do
+             :single_nalu ->
+               buffer_output(payload, buffer)
+
+             :fu_a ->
+               handle_fu(header, rest, buffer, state)
+
+             :stap_a ->
+               handle_stap(rest, buffer)
+           end)
+    ~>> ({:error, _} ->
+           log_malformed_buffer(buffer)
+           {{:ok, redemand: :output}, %State{state | pp_acc: nil}})
+  end
+
+  @impl true
+  def handle_demand(_output_pad, size, :buffers, _ctx, state),
+    do: {{:ok, demand: {:input, size}}, state}
+
+  def handle_demand(_, _, :bytes, state), do: {{:error, :not_supported_unit}, state}
+
+  @impl true
+  def handle_event(:input, %Discontinuity{}, _context, %State{pp_acc: %FU{}} = state),
+    do: {:ok, %State{state | pp_acc: nil}}
+
+  def handle_event(_pad, event, _context, state), do: {{:ok, forward: event}, state}
+
+  defp handle_fu(header, data, %Buffer{metadata: metadata} = buffer, state) do
+    %{rtp: %{sequence_number: seq_num}} = metadata
+
+    FU.parse(data, seq_num, map_state_to_fu(state))
+    ~>> (
```

Comment:
It could be a case with `{:error, _} = error` pass through, but I think that is the place where `~>>` is actually useful.

Reply:
It would be something like this:
```elixir
case parse(...) do
  {:ok, smth} -> do_smth(smth)
  {:incomplete, othersmth} -> do_other_smth(othersmth)
  default -> default
end
```

Which is very similar code what `~>>` would compile to.